augur/metadata_file.py

```diff
@@ -0,0 +1,89 @@
+import functools
+from os import access, R_OK
+import pandas
+
+
+class MetadataFile:
+    """
+    Represents a CSV or TSV file containing metadata
+
+    The file must contain exactly one of a column named `strain` or `name`,
+    which is used to match metadata with samples.
+    """
+
+    def __init__(self, fname, query):
+        self.fname = fname
+        self.query = query
+
+        self.key_type = self.find_key_type()
+
+        self.check_file()
+
+    def check_file(self):
+        if not access(self.fname, R_OK):
+            raise ValueError(
+                f"Metadata file {self.fname} does not exist or is not readable"
+            )
```

Comment:
This method does not seem to be necessary and it isn't being used as maybe you'd expect. The preceding `find_key_type` call already tries to open the file when it loads the columns and it's this call that allows the `FileNotFoundError` to be raised as expected in the unit tests before the `check_file` method is called.

I prefer the current behavior where `check_file` doesn't get called, since the `FileNotFoundError` is more specific than the `ValueError`.

Reply:
Removed.